hca/util/__init__.py

```diff
@@ -447,14 +446,24 @@ def _get_jwt_from_service_account_credentials(self):
                                 algorithm='RS256').decode()
         return signed_jwt, exp
 
+    def expired_token(self):
+        """Return True if we have an active session containing an expired (or nearly expired) token."""
+        ten_second_buffer = 10
+        if self._authenticated_session:
+            token_expiration = self._authenticated_session.token.get('expires_at', None)
+            if token_expiration:
+                if token_expiration <= time.time() + ten_second_buffer:
```

Comment:
What is the rationale for the time buffer versus just letting the token expire? And why 10 seconds?

Reply:
I though it should have _some_ buffer, since I can imagine checking a token and finding it valid with 0.001 seconds left, but when you go to actually make the call it errors because it's expired in that 0.001 seconds.  10 seconds is overkill, but the default expiration is an hour so I figured refreshing 10 seconds early is a comparably negligible buffer with no loss of efficiency that should avoid the situation I described.